app/controllers/api/v2/registration_commands_controller.rb

```diff
@@ -30,6 +33,10 @@ def create
           return
         end
 
+        if params[:repo] || params[:repo_gpg_key_url]
+          Foreman::Deprecation.api_deprecation_warning("Use repo_data parameter instead of repo and repo_gpg_key_url. They will be removed, soon.")
```

Comment:
I think we should keep the original functionality, meaning that when a user calls the API with the `repo` and `repo_gpg_key_url` parameters, it adds the repository to the `repo_data` variable and continues with the flow.

Reply:
Oh wait a sec, I see it below :)